ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -582,10 +586,7 @@ private void DetachViewFromInstance(ReactRootView rootView, IReactInstance react
 
             DispatcherHelpers.AssertOnDispatcher();
 
-            if (_lifecycleState == LifecycleState.Resumed)
-            {
-                reactContext.OnSuspend();
-            }
+            _lifecycleStateMachine.SetContext(null);
 
```

Comment:
Did we lose an `OnSuspend` call here? Sorry @reseul - I know the refactoring makes the diff a bit more complicated.

Reply:
I moved the logic inside SetContext. State machine brings a new context up to the current state, and a leaving context to OnSuspend.

---
In reply to: [166249288](https://github.com/Microsoft/react-native-windows/pull/1616#discussion_r166249288) [](ancestors = 166249288)